### PR TITLE
DEP: Remove Python2 specific C module setup

### PR DESCRIPTION
**Pull request by sethtroisi**
I'm excited about cleaning up all the Python2 specific code.

I broke out this small change first to make sure this can go in now and is a wanted changed.
If so there are ~700 instances of NPY_PY3K at least 50 of which contain real logic.

### CODE REVIEW

**Comment by mattip** [comment]
This will have to wait until 2020 when we drop python2 support. See the other PRs with the `Py3K` label. xref [relevant comment](https://github.com/numpy/numpy/pull/14376#issuecomment-525283845)

**Comment by sethtroisi** [comment]
Okay, happy to wait :)
This making backport for 1.16 harder makes a lot of sense

File: numpy/core/src/umath/_umath_tests.c.src [hunk 1]
```diff
@@ -598,25 +597,15 @@ static struct PyModuleDef moduledef = {
         NULL,
         NULL
 };
-#endif
 
 /* Initialization function for the module */
-#if defined(NPY_PY3K)
 #define RETVAL(x) x
```
Comment (jdufresne) [inline numpy/core/src/umath/_umath_tests.c.src]: `RETVAL` is now a pointless alias. Depending on how much should be done in one go, could potentially remove it as well.

Comment (sethtroisi) [reply numpy/core/src/umath/_umath_tests.c.src]: I'm planning to clean it up in a 2nd pass.

**Comment by sethtroisi** [comment]
It's after 2020! Can we start submitting Py3K fixes?

**Comment by charris** [comment]
Yes, it is time. Lets take a look at the pending versions and take them in sequence. I think a rebase will fix the CircleCI  failure.

**Comment by sethtroisi** [comment]
I rebased, but circleci failed again during building devdocs, I don't touch any files there so I'm confused as to why.

**Comment by mattip** [comment]
xref #15227

**Comment by eric-wieser** [comment]
> I'm planning to clean it up in a 2nd pass.

This patch looks great to me, but let's remove `RETVAL` at the same time, since the macro makes no sense now that the ifdefs are gone.

**Comment by sethtroisi** [comment]
@eric-wieser I cleaned up RETVAL and some other uses of PY_VERSION_HEX and NPY_PY3K in #15231 it might be cleaner to just pull that (or to pull this then I can rebase #15231 immediately after)

**Comment by eric-wieser** [comment]
Closing this one in favor of #15231, as suggested by @seberg

**Pull request closed without merging**